metasynth/distribution/discrete.py

```diff
@@ -116,21 +109,20 @@ def draw(self):
                 self.key_set.add(random_number)
                 return random_number
 
-    def information_criterion(self, values):
-        vals = values[~np.isnan(values)]
-        if np.min(values) < self.low:
+    def _information_criterion(self, values):
+        if values.min() < self.low:
             return 3+999*len(values)
 
         # If the values are not unique the fit is extremely bad.
-        if len(set(vals)) != len(vals):
+        if len(set(values)) != len(values):
             return 3+999*len(values)
```

Comment:
This somehow seems sketchy. Do we still use this?

Reply:
Which part do you find sketchy? I do think this is still used to determine whether we need to send a warning.